src/mapper/pkg/otelexporter/otel_exporter.go

```diff
@@ -0,0 +1,140 @@
+package otelexporter
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/otterize/network-mapper/src/mapper/pkg/intentsstore"
+	sharedconfig "github.com/otterize/network-mapper/src/shared/config"
+	"github.com/otterize/network-mapper/src/shared/version"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+)
+
+type OtelExporter struct {
+	config        Config
+	counter       metric.Int64Counter
+	intentsHolder *intentsstore.IntentsHolder
+}
+
+func newResource() (*resource.Resource, error) {
+	return resource.Merge(resource.Default(),
+		resource.NewWithAttributes(semconv.SchemaURL,
+			semconv.OTelLibraryName("otterize/network-mapper"),
+			semconv.OTelLibraryVersion(version.Version()),
+		))
+}
+
+const DefaultMetricEndpoint = "ingest.lightstep.com:443"
+
+// uses same name as expected in opentelemetry-collector-contrib's servicegraphprocessor
+const CounterMetricName = "traces_service_graph_request_total"
+const ClientAttributeName = "client"
+const ServerAttributeName = "server"
+
+func getenv(key, fallback string) string {
+	value := os.Getenv(key)
+	if len(value) == 0 {
+		return fallback
+	}
+	return value
+}
+
+func newMeterProvider(ctx context.Context, res *resource.Resource, exportInterval time.Duration) (*sdk.MeterProvider, error) {
+	metricExporter, err := otlpmetricgrpc.New(ctx,
+		otlpmetricgrpc.WithEndpoint(getenv("OTEL_EXPORTER_OTLP_ENDPOINT", DefaultMetricEndpoint)),
```

Comment:
In the PR description, you wrote that the OTEL SDK could pick this up automatically from environment variables vs configured using code. This is preferable so we don't have `getenv`. If it's not possible, please use `viper` config to set the configuration key and its default value, and then use `viper.GetString` instead of `getenv`.

Reply:
Updated to prefer the default OTel SDK env vars (which are extensive): https://github.com/open-telemetry/opentelemetry-go/blob/0e874c3468c2cbc5bbaf075537be6c0f2152b5ef/CHANGELOG.md?plain=1#L1426